helm/kfp-operator/templates/_helpers.tpl

```diff
@@ -60,3 +60,26 @@ Only render object if not empty.
 {{ . | toYaml }}
 {{- end }}
 {{- end }}
+
+{{/*
+Populate configuration with fallbacks and overrides.
+*/}}
+{{- define "fallbackConfiguration" -}}
+workflowNamespace: {{ .Values.namespace.name }}
+{{- end }}
+{{- define "configurationOverrides" -}}
+providerConfigFile: provider.yaml
+workflowTemplatePrefix: {{ include "kfp-operator.fullname" . }}-
+{{- if .Values.manager.multiversion.enabled }}
+multiversion: true
+{{- end -}}
+{{- end }}
+
+{{- define "kfp-operator.configuration" -}}
+{{ merge (include "configurationOverrides" . | fromYaml) .Values.manager.configuration (include "fallbackConfiguration" . | fromYaml) | toYaml }}
+{{- end }}
+
+{{- define "kfp-operator.workflow.namespace" -}}
+{{- $config := (include "kfp-operator.configuration" . | fromYaml) -}}
```

Comment:
Can't decide what is less ugly between
- This
- Having a separate `kfp-operator.configuration` that is not in YAML
- `kfp-operator.configuration` never YAML and to `toYaml` when calling it

Reply:
Thanks to non-typed Helm, this is the only option.